### ISSUE TITLE
Deconvolution seems not been used in the project?

### ISSUE THREAD

**Issue by RockWenJJ** [InFoCusp/tf_cnnvis#46]
Hi, thanks for your valuable work! I have learned a lot from your codes.
I'm trying to implement the method described in "Visualizing and Understanding Convolutional Networks" by Matthew D. and they seemed to have used the deconvolution method.  But I couldn't find the deconvolution method in your codes even in the  _deconvolution function. The reconstruction seems to be gotten by tf.gradients() rather than the tf.nn.conv2d_transpose() which is the official deconvolution method.
`reconstruct = [tf.gradients(tf.transpose(tf.transpose(op_tensor)[featuremap[i]]), X)[0] for i in range(config["N"])]`
Although the reconstruction by your codes looks like to the results of that paper, I'm not sure whether or not you are using the same method. Can you explain a little bit about the differences?
Look forward to your reply!

**Comment by falaktheoptimist** [comment]
In TF, conv2d_transpose - which is the deconvolution operation is same as gradient. See: https://www.tensorflow.org/api_docs/python/tf/nn/conv2d_transpose 
Quoting from that page

> This operation is sometimes called "deconvolution" after Deconvolutional Networks, but is actually the transpose (gradient) of conv2d rather than an actual deconvolution.

So, essentially it's doing the same thing as mentioned in the paper.
